### ISSUE TITLE
Support for PHP 8

### ISSUE THREAD

**Issue by ls-renaud-tilte** [digiaonline/lumen-newrelic#31]
### What did you do?

Tried to install on `php:8.0-fpm-alpine`

### What did you expect to happen?

Successful install

### What actually happened?

```
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - nordsoftware/lumen-newrelic[3.0.0, ..., 3.1.1] require php ^7.2 -> your php version (8.0.1) does not satisfy that requirement.
    - Root composer.json requires nordsoftware/lumen-newrelic ^3.0 -> satisfiable by nordsoftware/lumen-newrelic[3.0.0, 3.1.0, 3.1.1].
```

### What versions of relevant language, framework and this module are you using?

Here's the basic `composer.json` I'm using.

```json
{
    "name": "laravel/lumen",
    "description": "The Laravel Lumen Framework.",
    "keywords": ["framework", "laravel", "lumen"],
    "license": "MIT",
    "type": "project",
    "require": {
        "php": "^8.0",
        "laravel/lumen-framework": "^8.0",
        "nordsoftware/lumen-newrelic": "^3.0",
        "zircote/swagger-php": "^3.0"
    },
    "require-dev": {
        "fzaninotto/faker": "^1.9.1",
        "phpunit/phpunit": "^9.3",
        "mockery/mockery": "^1.3.1"
    },
    "autoload": {
        "psr-4": {
            "App\\": "app/",
            "Database\\Factories\\": "database/factories/",
            "Database\\Seeders\\": "database/seeders/"
        }
    },
    "autoload-dev": {
        "classmap": [
            "tests/"
        ]
    },
    "scripts": {
        "post-root-package-install": [
            "@php -r \"file_exists('.env') || copy('.env.example', '.env');\""
        ]
    },
    "config": {
        "preferred-install": "dist",
        "sort-packages": true,
        "optimize-autoloader": true
    },
    "minimum-stability": "dev",
    "prefer-stable": true
}
```

PHP 8 doesn't bring anything that would break the current version of the package, the upgrade should be straight forward, something like

```json
"require": {
        "php": "^7.2 | ^8.0",
        "intouch/newrelic": "^1.0",
        "illuminate/support": "^5.5 | ^6.0 | ^7.0 | ^8.0",
        "nordsoftware/lumen-chained-exception-handler": "^2.0"
    },
```

May also require an upgrade of the `nordsoftware/lumen-chained-exception-handler` package.

**Comment by Jalle19** [comment]
Can you make a pull request?

**Comment by ls-renaud-tilte** [comment]
I'll fork the repo to test it properly, you can expect a PR in the next few days.

**Comment by ls-renaud-tilte** [comment]
Ok so, I updated the code for both repos and managed to everything up and running with PHP8 ... except for the New Relic agent. The NR team is still working on the agent and doesn't have a viable product yet, the only officially supported versions of PHP are up to 7.4.

I'll keep an eye on the NewRelic roadmap and deliveries, and will switch the PRs from draft to actual work when the agent is available.

source:
- [New Relic Engineering Board](https://github.com/orgs/newrelic/projects/18)
- [New Relic Roadmap](https://github.com/orgs/newrelic/projects/23)

**Comment by luizberger** [comment]
> Ok so, I updated the code for both repos and managed to everything up and running with PHP8 ... except for the New Relic agent. The NR team is still working on the agent and doesn't have a viable product yet, the only officially supported versions of PHP are up to 7.4.
> 
> I'll keep an eye on the NewRelic roadmap and deliveries, and will switch the PRs from draft to actual work when the agent is available.
> 
> source:
> 
> * [New Relic Engineering Board](https://github.com/orgs/newrelic/projects/18)
> * [New Relic Roadmap](https://github.com/orgs/newrelic/projects/23)

The NewRelic team [released](https://github.com/newrelic/newrelic-php-agent/issues/35#issuecomment-824298729) a new version of PHP Agent, adding support for PHP8

**Comment by neclimdul** [comment]
8 has been supported by NewRelic for a while now. Is it possible to get #32 merged?

**Comment by hugovk** [comment]
PR #32 (and PR https://github.com/digiaonline/lumen-chained-exception-handler/pull/11, which blocks it) drop support for PHP 7.2, but 7.2 has been EOL for a year now so I'm fine with dropping it. It's more important to support PHP 8.

| cycle | latest |  release   |  support   |    eol     |
|:------|:-------|:----------:|:----------:|:----------:|
| 8.0   | 8.0.12 | 2020-11-26 | 2022-11-26 | 2023-11-26 |
| 7.4   | 7.4.25 | 2019-11-28 | 2021-11-28 | 2022-11-28 |
| 7.3   | 7.3.32 | 2018-12-06 | 2020-12-06 | 2021-12-06 |
| 7.2   | 7.2.34 | 2017-11-30 | 2019-11-30 | 2020-11-30 |

If there's no objections from @Jalle19 then let's get them merged and out.

**Comment by Jalle19** [comment]
No objections from me

**Comment by hugovk** [comment]
Great!

https://github.com/digiaonline/lumen-chained-exception-handler/pull/11 has been merged and released as 3.0.0.

Next: https://github.com/digiaonline/lumen-newrelic/pull/32 can be updated.
